test-collateral/generate-policy/src/main.rs

```diff
@@ -0,0 +1,524 @@
+        .zip(&arguments.roles)
+        .enumerate()
+    {
+        if let Ok(mut file) = File::open(cert) {
+            let mut content = String::new();
+
+            file.read_to_string(&mut content)
+                .expect("Failed to read file.");
+
+            content = content.replace(
+                "-----BEGIN CERTIFICATE-----",
+                "-----BEGIN CERTIFICATE-----\\n",
+            );
+            content = content.replace("-----END CERTIFICATE", "\\n-----END CERTIFICATE");
+
+            let json = json!({
+                "cert": content,
+                "id": id,
+                "roles": roles,
+            });
+
+            values.push(json);
+        } else {
+            eprintln!("Could not open certificate file.");
+            exit(1);
+        }
+    }
+
+    json!(values)
+}
+
+/// Serializes the enclave server certificate expiry timepoint to a JSON value,
+/// computing the time when the certificate will expire as a point relative to
+/// the current time.
+fn serialize_enclave_certificate_expiry(arguments: &Arguments) -> Value {
+    let duration = Duration::hours(arguments.certificate_lifetime);
+    let expiry = Utc::now() + duration;
+
+    json!({
+        "year": expiry.year(),
+        "month": expiry.month(),
+        "day": expiry.day(),
+        "hour": expiry.hour(),
+        "minute": expiry.minute()
+    })
+}
+
+/// Serializes the Veracruz policy file as a JSON value.
+fn serialize_json(arguments: &Arguments) -> Value {
+    info!("Serializing JSON policy file.");
+
+    let identities = serialize_identities(arguments);
+
+    let sinaloa_url = format!("{}", &arguments.sinaloa_url.as_ref().unwrap());
+    let tabasco_url = format!("{}", &arguments.tabasco_url.as_ref().unwrap());
+
+    let enclave_cert_expiry = serialize_enclave_certificate_expiry(arguments);
+    let ciphersuite = POLICY_CIPHERSUITE;
+
+    let data_provision_order = "";
+    let streaming_order = "";
+
+    let pi_hash = compute_program_hash(arguments);
+    let debug = &arguments.debug;
+    let execution_strategy = &arguments.execution_strategy;
+
+    json!({
+        "identities": identities,
+        "sinaloa_url": sinaloa_url,
+        "enclave_cert_expiry": enclave_cert_expiry,
+        "ciphersuite": ciphersuite,
+        "tabasco_url": tabasco_url,
+        "data_provision_order": data_provision_order,
+        "streaming_order": streaming_order,
+        "pi_hash": pi_hash,
+        "debug": debug,
+        "execution_strategy": execution_strategy,
+    })
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Entry point.
+////////////////////////////////////////////////////////////////////////////////
+
+/// Entry point: reads the command line, serializes the policy file to JSON, and
+/// then writes the serialized JSON to the specified output file.
+fn main() {
+    let arguments = parse_command_line();
+
+    info!("Writing JSON file, {}.", arguments.output_policy_file);
```

Comment:
Just an idea, but it would be nice if this also printed the policy hash to stdout. That way a user could visual check that policies are correct, much like looking at git shas.

Reply:
Good idea, I will add this.